### DISCUSSION TITLE
what is required, if assets in the default library have been rotated externally? or if other metadata will be changed?

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#6219] [Q&A]
I rotated some photos, contained in the default library, using digiKam.
I also want to add or change geo data using external tools.
And I want immich to detect and accept these changes.

digiKam writes metadata changes back as embedded information into the pictures, additionally it creates (and changes) sidecar files. I started to create sidecar files especially for immich, because my understanding was, that immich can use these files and its content.

I don't get the rotation accepted by immich.
Furthermore, I tried different administrator jobs, without success.

How can I force immich to accept these changes?

Related to #6218, #1991, #1695

**Comment by bo0tzz** [comment]
After changing rotation metadata, you would need to:
1. (optionally) run the sidecar job to find sidecar files
2. Run the metadata extraction to read the new rotation
3. Run the thumbnail generation to generate new thumbs with the rotation applied.

**Comment by aisbergde** [comment]
OK, I did not run the "GENERATE THUMBNAILS" job for all assets. Doing this only for missing is not enough. Now it will regenerate for all assets (> 60.000). This will take some hours. And it means wasting resources.

It would be an improvement, if immich could be asked to regenerate thumbnails only for assets, where the metadata was changed. Or the thumbnail generation could be triggered for relevant assets, if metadata change has been detected (I think, only rotation could have influence).
Does a related feature request exist, is it on the roadmap, or should I create a feature request?

Another idea would be to regenerate thumbnails only per library. My default library contains only a few thousand assets.

Or is it a better way, to move these assets, which needs to be rotated, into an external library? It looks like the change detection in external libraries is much better. I had no issues to rotate them.

**Reply by bo0tzz** [reply]
You can regenerate thumbnails for specific assets from the gallery view.

**Reply by aisbergde** [reply]
that's a good solution! Thank you!

**Reply by aisbergde** [reply]
Unfortunately, it doesn't work. I try to refresh metadata and to refresh thumbnails in the one-asset-view. Nothing changes. But when I use the Synology file station or when I use Windows and a network drive, then the rotated picture looks correct. There could be something different.

Here is one test photo and its sidecar file

[immich-rotate-issue.zip](https://github.com/immich-app/immich/files/13850589/immich-rotate-issue.zip)

And I have one idea about a possible reason:
digikam not only changed the sidecar file, but also the photo itselfs. Especially, when rotating photos. And maybe digiKam rotated the picture itself, and now the metadata says: don't rotate the photo, because it is as needed.

Then the question is not only about refreshing metadata, but to reimport the rotated asset. And maybe immich doesn't do this, because immich assumption is, that assets in the default library will not change?

**Reply by aisbergde** [reply]
How it works in digiKam: https://docs.digikam.org/en/setup_application/metadata_settings.html#rotation-settings
And this is a setting for all collections. If these settings were per digiKam collection, I could say: "don't change the content in the immich default library". But I want to change the content in general in all other collections.

It looks like, I have to wait, until immich implements a content change detection also in the default library. Maybe I have to move assets to be rotated from default library to external library. But then I have duplicates, because offline detection is missing in the default library. It is not so easy to solve :-)
